src/cachecheck.cpp

```diff
@@ -79,29 +79,26 @@ void CheckCaches()
 		rs->GetEntry(DIAGDIR_NW)->CheckIntegrity(rs);
 	}
 
+	std::vector<NewGRFCache> grf_cache;
+	std::vector<VehicleCache> veh_cache;
+	std::vector<GroundVehicleCache> gro_cache;
+	std::vector<TrainCache> tra_cache;
+
 	for (Vehicle *v : Vehicle::Iterate()) {
 		if (v != v->First() || v->vehstatus & VS_CRASHED || !v->IsPrimaryVehicle()) continue;
 
 		uint length = 0;
```

Comment:
Shouldn't this move to line 117? And can't the `length++` at line 106 be removed?

Reply:
Yes. I just hadn't spotted this in the excitement of removing the first loop :-)